### DISCUSSION TITLE
How to take a bean as an argument of constructor of another bean on Karavan?

### DISCUSSION THREAD

**Discussion by MN-Dheeraj** [apache/camel-karavan#1251] [Q&A] [answered]
### Ask a question

I am creating beans on camel-karavan.

Please find my beans below.

- beans:

      - constructors:
        "0": oracle.jdbc.driver.OracleDriver
        "1": my DB's address
        "2": my DB's username
        "3": my DB's password
      name: dataSource
      properties:
        driverClass: oracle.jdbc.driver.OracleDriver
        url: my DB's address
        username: my DB's username
        password: my DB's password
      type: org.springframework.jdbc.datasource.SimpleDriverDataSource
    - constructors:
        "0": "\"#bean:dataSource\""
      name: esbJdbcTemplate
      type: org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate

**Issue:**
My bean 'dataSource' instantiated successfully, though it's type is of spring class.
But, the other bean 'esbJdbcTemplate' of type 'org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate' couldn't be instantiated.
The constructor of NamedParameterJdbcTemplate takes one argument which is 'dataSource'. But, I see this error.

org.apache.camel.RuntimeCamelException: Error pre-parsing resource: file:dheeraj-il.camel.yaml
        at org.apache.camel.dsl.yaml.YamlRoutesBuilderLoader.preParseRoute(YamlRoutesBuilderLoader.java:975)  
 ................
Caused by: java.lang.RuntimeException: Error creating bean: esbJdbcTemplate of type: #class:org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
        at org.apache.camel.dsl.yaml.deserializers.BeansDeserializer.registerBean(BeansDeserializer.java:212) 
  ................
Caused by: java.lang.IllegalStateException: Cannot create instance of class: org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
        at org.apache.camel.support.PropertyBindingSupport.resolveBean(PropertyBindingSupport.java:1598) 

Is there anything wrong with my definition? I didn't find how to take bean as an argument to a constructor at this doc : https://camel.apache.org/components/4.4.x/others/yaml-dsl.html

### Variant

Vs Code Extension

### Container Management (if applicable)

None

### Operating System (if applicable)

Windows

### Version

4.4.0

**Comment by davsclaus** [comment, marked as answer]
Yes this requires to be supported in Apache Camel to make this possible in Karavan. This will be implemented in Camel 4.7.0 onwards.
https://issues.apache.org/jira/browse/CAMEL-20745

**Reply by jabhishek01** [reply]
@davsclaus Can you please suggest if there is any way to use encrypted password for datasource bean ? Is there any support ?

**Reply by davsclaus** [reply]
Yes study Camel it has vaults or camel-jasypt etc.

**Reply by jabhishek01** [reply]
okay but camel-jasypt didnt work for us with camel-karavan 4.6.0 when we run project route in dev mode. It printed same value that we specified as encrypted value.. Decryption didnt worked . Is there specific version to target ?
